landing-page/content/common/spark-quickstart.md

```diff
@@ -0,0 +1,379 @@
+WHEN MATCHED THEN UPDATE SET t.Fare_amount = t.Fare_amount + u.Fare_amount
+WHEN NOT MATCHED THEN INSERT *;
+```
+Querying the `taxis` table, you can see that the fare amount of `0.50` has been added to `15.32` for `VendorID = 1 AND TripID = 1000371`.
+```sql
+SELECT VendorID, TripID, Fare_amount
+FROM prod.nyc.taxis
+WHERE VendorID = 1 AND TripID = 1000371;
+```
+| VendorID | TripID  | Fare_amount |
+| -------- | ------- | ----------- |
+| 1        | 1000371 | 15.82       |
+
+{{< hint warning >}}
+Iceberg supports writing DataFrames using the new [v2 DataFrame write API](../spark-writes#writing-with-dataframes). The old `write` API is supported, but _not_ recommended.
+{{< /hint >}}
+
+## Adding Iceberg to your Existing Spark Environment
+
+To add Iceberg, use the `--packages` option.
+
+{{% codetabs "AddIcebergToSpark" %}}
+{{% addtab "SparkSQL" checked %}}
+{{% addtab "SparkShell" %}}
+{{% addtab "PySpark" %}}
+{{% tabcontent "SparkSQL"  %}}
+```sh
+spark-sql --packages org.apache.iceberg:iceberg-spark-runtime-3.2_2.12:{{% icebergVersion %}}
+```
+{{% /tabcontent %}}
+{{% tabcontent "SparkShell" %}}
+```sh
+spark-shell --packages org.apache.iceberg:iceberg-spark-runtime-3.2_2.12:{{% icebergVersion %}}
+```
+{{% /tabcontent %}}
+{{% tabcontent "PySpark" %}}
+```sh
+pyspark --packages org.apache.iceberg:iceberg-spark-runtime-3.2_2.12:{{% icebergVersion %}}
+```
+{{% /tabcontent %}}
+{{% /codetabs %}}
+
+{{< hint info >}}
+If you want to include Iceberg in your Spark installation, add the [`iceberg-spark-runtime-3.2_2.12` Jar](spark-runtime-jar) to Spark's `jars` folder.
+{{< /hint >}}
+
+[spark-runtime-jar]: https://search.maven.org/remotecontent?filepath=org/apache/iceberg/iceberg-spark-runtime-3.2_2.12/{{% icebergVersion %}}/iceberg-spark-runtime-3.2_2.12-{{% icebergVersion %}}.jar
+
+### Adding catalogs
+
+Iceberg comes with [catalogs](../spark-configuration#catalogs) that enable SQL commands to manage tables and load them by name. Catalogs are configured using properties under `spark.sql.catalog.(catalog_name)`.
+
+This command creates a path-based catalog named `local` for tables under `$PWD/warehouse` and adds support for Iceberg tables to Spark's built-in catalog:
+
+
+{{% codetabs "AddingACatalog" %}}
+{{% addtab "SparkSQL" checked %}}
+{{% addtab "SparkShell" %}}
+{{% addtab "PySpark" %}}
+{{% tabcontent "SparkSQL"  %}}
+```sh
+spark-sql --packages org.apache.iceberg:iceberg-spark-runtime-3.2_2.12:{{% icebergVersion %}}\
+    --conf spark.sql.extensions=org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions \
+    --conf spark.sql.catalog.spark_catalog=org.apache.iceberg.spark.SparkSessionCatalog \
+    --conf spark.sql.catalog.spark_catalog.type=hive \
+    --conf spark.sql.catalog.prod=org.apache.iceberg.spark.SparkCatalog \
+    --conf spark.sql.catalog.prod.type=hadoop \
+    --conf spark.sql.catalog.prod.warehouse=$PWD/warehouse \
+    --conf spark.sql.defaultCatalog=prod
```

Comment:
Are these commands compatible with the docker-compose setup?

Maybe we should just give the config properties and have a way to set config in the docker image?

Reply:
Yes they're compatible with the docker-compose setup. I wanted to include them this way incase someone wants to add a spark environment to their own spark environment (an engineer trying Iceberg at their job which already has a spark cluster for example). I updated the title of this section to "Adding A Catalog to Your Own Spark Environment" to be more clear about the purpose of this section.